libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -371,57 +373,66 @@ bool AP_ExternalAHRS_VectorNav::decode(char c)
 }
 
 // decode the most recently consumed term
-// returns true if new sentence has just passed checksum test and is validated
+// returns true if new term is valid
 bool AP_ExternalAHRS_VectorNav::decode_latest_term()
 {
+    // Check the first two terms (In most cases header + reg number) that they match the sent
+    // message. If not, the response is invalid.
     switch (nmea.term_number) {
         case 0:
-            if (strcmp(nmea.term, "VNRRG") != 0) {
+            if (strncmp(nmea.term, "VNERR", nmea.term_offset) == 0) {
+                nmea.error_response = true;  // Message will be printed on next term 
+            } else if (strncmp(nmea.term, message_to_send, nmea.term_offset) != 0) {
                 return false;
             }
-            break;
-
-        case 1:
-            if (nmea.register_number != strtoul(nmea.term, nullptr, 10)) {
+            return true;
+        case 1: 
+            if (nmea.error_response) {
+                GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "VectorNav received VNERR code: %s", nmea.term);
+            } else if (strncmp(nmea.term, message_to_send + 6,
```

Comment:
```suggestion
            } else if (strncmp(nmea.term, MIN(message_to_send + 6, strlen(message_to_send))
```

I'm not a fan of the structure of this driver, and I haven't even really understood it yet.  But the code that's currently in here looks fall-out-of-sky dangerous here.  It's probably not possible to get in here with `strlen(message_to_send)` less than 6, but if we *do* get here then bad things may happen.

Reply:
@peterbarker 

It should never be possible to hit this point with length < 6 (All responses must begin with "VNXXX,", and to hit the term==1, there has to have been something received after the comma)

I'm also not clear that it would be particularly dangerous, would it not just worst case compare to unutilized memory allocated to message_to_send? Of course I'm assuming message_to_send capacity is greater than 6 + nmea.term_offset